ONSdigital/takeon-business-layer
# Feature/spp 888 logging

# Description
Added structured logging to the Business Layer (Java) with all the custom levels which includes TRACE, DEBUG, INFO, ERROR, WARN, ALL and OFF level.
https://collaborate2.ons.gov.uk/jira/browse/SPP-888
I included all the levels of logging in the controller and entity classes

# How to test?
The changes can be tested in SPP1515 environment and perform the following steps
Repository: takeon-business-layer
1. Modify LOG4J2.xml in the location - src/main/resources/log4j2.xml 
2. Change the levels of the log in the xml file based on project requirement. For example resolving any production issue, switch back to DEBUG level and normal root level is INFO
3. Commit and push the change to the branch
4. This will build and deploy the business layer to Kubernetes cluster
5. Run the application and verify the logs in
AWS Management Console -> CloudWatch -> Logs -> Log groups -> /aws/containerinsights/spp-es-takeon-spp1515-eks-cluster/application
6. Verify the Log streams and you can see a file with 'takeon_business_layer' 
7. Verify the level of log in the file based on switching on and off of log levels in Log4J2.xml


# Links
# Different Levels of Java logs
https://www.tutorialspoint.com/log4j/log4j_logging_levels.htm
# Previous Spike
https://collaborate2.ons.gov.uk/confluence/display/ESD/Basic+Logging+using+Log4j2+in+Spring+Boot
